cmd/telemeter-server/main.go

```diff
@@ -84,18 +82,33 @@ func main() {
 		MemcachedInterval:  10,
 		TenantID:           "FB870BF3-9F3A-44FF-9BF7-D7A047A52F43",
 	}
+}
+
+func main() {
+	opt := defaultOpts()
+
+	var listen, listenInternal string
 	cmd := &cobra.Command{
 		Short:         "Aggregate federated metrics pushes",
 		Long:          desc,
 		SilenceErrors: true,
 		SilenceUsage:  true,
 		RunE: func(cmd *cobra.Command, args []string) error {
-			return opt.Run()
+			listener, err := net.Listen("tcp", listen)
```

Comment:
all these changes look ok but they are unrelated to the fix. let's merge them in another PR so that we can keep the changes small and easy to reason about